visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -25,50 +71,351 @@ export class LoadInitialFileService {
 		private httpClient: HttpClient
 	) {}
 
-	async loadFileOrSample() {
+	async loadFilesOrSampleFiles() {
+		const isFileQueryParameterPresent = this.checkFileQueryParameterPresent()
+		await (isFileQueryParameterPresent ? this.loadFilesFromQueryParams() : this.loadFilesFromIndexedDB())
+	}
+
+	checkFileQueryParameterPresent() {
+		return Boolean(this.urlUtils.getParameterByName("file"))
+	}
+
+	private async loadFilesFromQueryParams() {
 		try {
-			const data = await this.urlUtils.getFileDataFromQueryParam()
-			this.tryLoadingFiles(data)
-			this.setRenderStateFromUrl()
+			const urlNameDataPairs = await this.urlUtils.getFileDataFromQueryParam()
+			const savedCcState = await readCcState()
+			if (!savedCcState) {
+				this.loadFileService.loadFiles(urlNameDataPairs)
+				this.setRenderStateFromUrl()
+				return
+			}
+
+			const savedFileStates = savedCcState.files
+			const savedNameDataPairs = savedFileStates.map(fileState => getNameDataPair(fileState.file))
+			const urlNameDataPairCheckSums = urlNameDataPairs.map(urlNameDataPair => urlNameDataPair.content.fileChecksum)
+			const savedNameDataPairCheckSums = savedNameDataPairs.map(savedNameDataPair => savedNameDataPair.content.fileChecksum)
+			if (stringify(urlNameDataPairCheckSums) === stringify(savedNameDataPairCheckSums)) {
+				this.applyAppSettings(savedCcState.appSettings)
+				this.loadFileService.loadFiles(savedNameDataPairs)
+				this.store.dispatch(setFiles({ value: savedFileStates }))
+				this.applyAllSettings(savedCcState)
+				this.setRenderStateFromUrl()
+			} else {
+				this.applyAllSettings(savedCcState)
+				this.loadFileService.loadFiles(urlNameDataPairs)
+				this.setRenderStateFromUrl()
+			}
```

Comment:
This could be extracted into to properly named methods to make the code more readable.